docker/package/baking_wizard.py

```diff
@@ -0,0 +1,303 @@
+#!/usr/bin/env python3
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+"""
+A wizard utility to help set up tezos-baker.
+
+Asks questions, validates answers, and executes the appropriate steps using the final configuration.
+"""
+
+import os, shlex, subprocess
+import readline
+from typing import List
+
+# Input validators
+
+
+def enum_validator(options: List[str]):
+    def _validator(input):
+        if not input in options:
+            raise ValueError(
+                "Please choose one of the provided values: " + ", ".join(options)
+            )
+
+    return _validator
+
+
+def filepath_validator(input):
+    if input and not os.path.isfile(input):
+        raise ValueError("Please input a valid file path.")
+
+
+def required_field_validator(input):
+    if not input.strip():
+        raise ValueError("Please provide this required option.")
+
+
+# Input completers
+
+
+def enum_completer(options: List[str]):
+    def _completer(input, state):
+        prefixed = [o for o in options if o.startswith(input)]
+        return prefixed[state]
+
+    return _completer
+
+
+# Wizard CLI utility
+
+
+def proc_call(cmd):
+    return subprocess.check_call(shlex.split(cmd))
+
+
+modes = ["baker", "node"]
+
+
+class Step:
+    def __init__(
+        self,
+        id: str,
+        prompt: str,
+        help: str,
+        default: str = None,
+        validator=None,
+        completer=None,
+        reqs=lambda _vs: True,
+    ):
+        self.id = id
+        self.prompt = prompt
+        self.help = help
+        self.default = default
+        self.validator = validator
+        self.completer = completer
+        self.reqs = reqs
+
+    def validate(self, input):
+        if self.validator is not None:
+            if isinstance(self.validator, list):
```

Comment:
Looks kinda sus that `Step` should know something about validator structure. Let's move validation logic from the `Step` class, so that it simply calls something like `self.validator.validate(input)`